### PR TITLE
bump viewer-crd-controller rock to ckf-1.8 (kfp 2.0.3)

### LINKED ISSUE #48: Update rockraft file of viewer-crd-controller for CKF 1.8
### What needs to get done

Update the corresponding `rockcraft.yaml` file to make sure the version of CKF 1.8 is built. 

DoD:
- The `rockcraft.yaml` is updated
- There is automation that builds the image in each PR
- There is automation that pushes the image, once a PR is merged

### Why it needs to get done

To make sure we have less CVEs in 1.8 than upstream, and enable us to be able to potentially create patches for package versions in the future

### PR DESCRIPTION
**Pull request by orfeas-k**
- bump source version from `2.0.0-alpha.7` to `2.0.3`
- update according to canonical/bundle-kubeflow#747
- rename base to ubuntu@20.04 as per new `rockcraft` syntax
- use `1.20/stable` snap for Go
- refactor `tox.ini` according to canonical/oidc-authservice-rock#14 and canonical/bundle-kubeflow#763
- use placeholder instead of bundle-integration tests due to #61 (there are no charm integration tests for this charm)
- add placeholder instead of sanity tests since there are none at the moment

Note CI needs a `sanity` and `integration` environment in order to not fail and be able to publish the produced ROCK after merging this.

### Testing
ROCK can be built and unit goes to active using it. I didn't run any hands-on tests since bundle-integration and manual tests will be run once all kfp ROCKs are updated.

Closes #48

### CODE REVIEW

File: viewer-crd-controller/tox.ini [hunk 1]
```diff
@@ -3,6 +3,7 @@
 [tox]
 skipsdist = True
 skip_missing_interpreters = True
+envlist = unit, sanity, integration
```
Comment (DnPlas) [inline viewer-crd-controller/tox.ini:6]: In your file you've got: `integration, sanity, export-to-docker, pack`. I think You want to update this envlist.

**Review by DnPlas — COMMENTED** [review]
Just a tiny little detail, otherwise LGTM

File: viewer-crd-controller/tox.ini (hunk 1 above)
Comment (orfeas-k) [reply viewer-crd-controller/tox.ini]: Updated! I 'm wondering how `tox` uses those, since it's working without those configured properly.

Comment (orfeas-k) [reply viewer-crd-controller/tox.ini]: Got it 
> `envlist` is a list of all environments that would be run if you invoke tox without the -e option.

File: viewer-crd-controller/tox.ini [hunk 2]
```diff
@@ -12,57 +13,61 @@ setenv =
...
+    ; git clone --branch {env:CHARM_BRANCH} {env:CHARM_REPO} {env:LOCAL_CHARM_DIR}
+    ; # upload rock to docker and microk8s cache, replace charm's container with local rock reference
+    ; bash -c 'NAME=$(yq eval .name rockcraft.yaml) && \
+    ;          VERSION=$(yq eval .version rockcraft.yaml) && \
+    ;          DOCKER_IMAGE=$NAME:$VERSION && \
+    ;          docker save $DOCKER_IMAGE > $DOCKER_IMAGE.tar && \
+    ;          sudo microk8s ctr image import $DOCKER_IMAGE.tar --digests=true && \
+    ;          yq e -i ".resources.kfp-viewer-image.upstream-source=\"$DOCKER_IMAGE\"" {env:LOCAL_CHARM_DIR}/charms/kfp-viewer/metadata.yaml'
+    ; # run bundle integration tests with rock
+    ; tox -c {env:LOCAL_CHARM_DIR} -e bundle-integration-v2 -- --model kubeflow
```
[the remark this answers is not in the document]
Comment (orfeas-k) [reply viewer-crd-controller/tox.ini:73]: in order to not having to rewrite this once bundle-integration tests work. Atm, they fail not because of ROCK functionality but for a different reason, which we need to troubleshoot https://github.com/canonical/pipelines-rocks/issues/61 but is out of scope for this effort

[the remark this answers is not in the document]
Comment (misohu) [reply viewer-crd-controller/tox.ini:73]: Can we add a little comment there

[the remark this answers is not in the document]
Comment (orfeas-k) [reply viewer-crd-controller/tox.ini]: Added

**Review by misohu — APPROVED** [review]
